babynames.py

```diff
@@ -45,9 +45,29 @@ def extract_names(filename):
     with the year string followed by the name-rank strings in alphabetical order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
-    names = []
-    # +++your code here+++
-    return names
+    with open(filename) as f:
+        open_file = f.read()
+    year_codex = r'Popularity\sin\s(\d\d\d\d)'
+    year = re.findall(year_codex, open_file)
+    print(year[0])
+    names_codex = r'<td>(\d+)</td><td>(\w+)</td><td>(\w+)</td>'
+    names_found = re.findall(names_codex, open_file)
+    print(names_found)
+    names = [year[0]]
+    names_dict = {}
+    for rank, boy_name, girl_name in names_found:
+        boy_entry = boy_name + " " + rank
```

Comment:
I think the layout of the names_dict will serve you better if you use just the names (boy or girl) as the keys, and the ranks (integers) as the values.  Then you will have an easier time manipulating the data.

Reply:
i like how you did the "rank, boy_name, girl_name in names_found".  Clever use of the groups there.  Like piero said, you may want to just flip-flop your dictionary.  Right now you have the keys as the rank and the values as the names.  Instead, try making the keys be the names, and the values be the rank.  Then, since you are running from #1 to #1000, it will not overwrite with larger values if you only add the new items not already in the dictionary.